File: projekt2/src/proj/java/model/plant/NightShade.java
package proj.java.model.plant;

import java.awt.Color;

import proj.java.model.Organism;
import proj.java.model.Plant;
import proj.java.model.World;

public class NightShade extends Plant{

  private static final int NIGHTSHADE_STRENGTH = 99;

  @Override
  protected void spread(int newRow, int newCol, World world) {
      world.setOrganism(newRow, newCol, new NightShade());
  }

  @Override
  public int getStrength() {
      return NIGHTSHADE_STRENGTH;
  }

  @Override
  public Color getColor() {
      return Color.BLUE;
  }

  @Override
  public boolean specialAbility(Organism other, World world, boolean specialAblityStopsMovement) {
      world.removeOrganism(other.getRow(), other.getCol());
      world.removeOrganism(this.getRow(), this.getCol());
      String message = String.format("%s zjada wilcze jagody i umiera" , other.getName());
      world.addMessage(message);
      specialAblityStopsMovement = true;
      return true;
  }
}

File: projekt3/NightShade.py
from Plant import *

class NightShade(Plant):
    __NIGHTSHADE_STRENGTH__ = 99
    __COLOR__ = "Blue"
    
    def __init__(self, row, col, world):
        super().__init__(row, col, self.__NIGHTSHADE_STRENGTH__,world)
    
    def getColor(self):
        return self.__COLOR__
      
    def getName(self):
      return "NightShade"

    def spread(self, newRow, newCol):
      self._world_.setOrganism(newRow, newCol, NightShade(newRow , newCol, self._world_))

    def specialAbility(self, other):
      self._world_.removeOrganism(other.getRow(), other.getCol())
      other.die()
      self._world_.removeOrganism(self.getRow(), self.getCol())
      self.die()
      event = other.getName() + " zjada wilcze jagody i umiera"
      self._world_.addEvent(event)
      self._world_.removeOrganism(self.getRow(), self.getCol())
      return True